salt/modules/restartcheck.py

```diff
@@ -298,6 +300,13 @@ def _kernel_versions_redhat():
     return kernel_versions
 
 
+def _is_older_nilrt():
+    '''
+    If this is an older version of NILinuxRT, return True. Otherwise, return False.
+    '''
+    return os.path.exists('/usr/local/natinst/bin/nisafemodeversion')
```

Comment:
This is strange. You're judging _version_ by just a mere place of the binary installation. Why you're so sure that the `nisafemodeversion` isn't up to the latest _version_ but just is a symlink for backward compatibility? If you want to look for version, then look for the version by actually calling it or refer to some data that would actually fetch the real version and compare it.

Reply:
We have two big incompatible versions (usually we call them older and newer NILRT) of the distro supported side by side and there is no way to distinguish them from the rootfs (no versioning info to make that distinction in any package). I could implement a machanism to do that but we'd still have to support already released images...

That nisafemodeversion file is not present t all in newer NILRT, as we have a unified bootflow with no "safe mode", so even though it looks ugly that's the most reliable way I could come up to distinguish older vs newer :sweat_smile:

Reply:
@10ne1 I hear you. It is still would be nicer if your distro would start using `/etc/os-release` like everyone else are doing.